transaction/src/manifest/generator.rs

```diff
@@ -1517,11 +1708,188 @@ where
     Ok(result)
 }
 
+pub fn generator_error_diagnostics(
+    s: &str,
+    err: GeneratorError,
+    style: CompileErrorDiagnosticsStyle,
+) -> String {
+    let (title, label) = match err.error_kind {
+        GeneratorErrorKind::InvalidAstType {
+            expected_type,
+            actual,
+        } => {
+            let title = format!(
+                "expected value kind {}, found {}",
+                expected_type.value_kind(),
+                actual.value_kind()
+            );
+            let label = format!(
+                "expected {}, found {}",
+                expected_type.value_kind(),
+                actual.value_kind()
+            );
+            (title, label)
+        }
+        GeneratorErrorKind::InvalidAstValue {
+            expected_type,
+            actual,
+        } => {
+            let mut types: Vec<String> = vec![];
+
+            // Remove potential duplicates.
+            // Duplicates might occure due to aliases,
+            // eg. PackageAddress, ComponentAddress are both Address
+            for ty in expected_type.iter() {
+                let ty = ty.value_kind().to_string();
+                if !types.contains(&ty) {
+                    types.push(ty)
+                }
+            }
+            let title = format!(
+                "expected value kind {}, found {}",
+                types.join(" or "),
+                actual.value_kind()
+            );
+            let label = format!(
+                "expected {}, found {}",
+                types.join(" or "),
+                actual.value_kind()
+            );
+            (title, label)
+        }
+        GeneratorErrorKind::UnexpectedValue {
+            expected_type,
+            actual,
+        } => {
+            // TODO: Consider better messages for aliases (eg. Bytes)
+            let title = format!(
+                "expected value kind {}, found {}",
```

Comment:
nitpick: Ditto, here we're not expecting a value kind, we're expecting a value. Probably better is `expected {} value, found {} value`

Reply:
Done